pkg/cmd/pipeline/start_test.go

```diff
@@ -454,6 +454,22 @@ func TestPipelineStart_ExecuteCommand(t *testing.T) {
 			wantError:  false,
 			goldenFile: true,
 		},
+		{
+			name: "Dry Run with use param defaults and output=json",
+			command: []string{"start", "test-pipeline",
+				"-s=svc1",
+				"-r=source=scaffold-git",
+				"--use-param-defaults",
+				"-l=jemange=desfrites",
+				"-n", "ns",
+				"--dry-run",
+				"--output", "json",
```

Comment:
You can remove `--output json` as this is sufficiently covered by other tests, but think the test idea is great.

Reply:
I wanted to test that the default value are actually getting populated or not, and that is why json output is matched with the golden file I created and they actually get populated, tests get passed :).

Reply:
Right, but not using `output` `json` will just use the default format of yaml instead, so it should still produce a golden file with appropriate output.

Reply:
Then I have to change the golden file as well .. sure will do that ..